### PR TITLE
Includes ast.Node.ContentEquals method

### PR DESCRIPTION
**Pull request by andrewemeryanz**
Included to test equality of node content independently of the
scanners used to parse them. Included to provide alternative approach
to Scanner.StripSource for testing equality.

### CODE REVIEW

File: ast/node.go [hunk 1]
```diff
@@ -176,3 +177,53 @@ func (n Branch) narrow() bool {
...
+		for k, v := range b {
+			switch v.(type) {
+			case One:
+				if !v.(One).Node.ContentEquals(other.One(k)) {
+					return false
+				}
+			case Many:
+				nodes := v.(Many)
+				otherNodes := other.Many(k) // may be nil
+				if otherNodes == nil || len(nodes) != len(otherNodes) {
```
Comment (ghost) [inline ast/node.go]: comment and nil check unnecessary. `len(nil) == 0`

File: ast/node.go [hunk 2]
```diff
@@ -176,3 +177,53 @@ func (n Branch) narrow() bool {
...
+func (l Leaf) ContentEquals(other Node) bool {
+	switch other.(type) {
+	case Leaf:
+		return l.Scanner().String() == other.(Leaf).Scanner().String()
+	}
+	return false
+}
+
+func (b Branch) ContentEquals(other Node) bool {
+	switch other.(type) {
```
Comment (ghost) [inline ast/node.go]: `switch other := other.(type)` so you don't need to the same thing in line 192

Same in the Leaf and Extra switches

**Review by ghost — CHANGES_REQUESTED** [review]
Minor changes requested.

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Please reply to all comments in the future

**Review by marcelocantos — APPROVED** [review]

**Pull request merged**